clickhouse-http-client/src/test/java/com/clickhouse/client/http/ClickHouseHttpClientTest.java

```diff
@@ -143,65 +144,109 @@ public void testUserAgent() throws Exception {
         final String sql = "select :uuid(String)";
 
         String uuid = UUID.randomUUID().toString();
+        String queryId = UUID.randomUUID().toString();
         try (ClickHouseClient client = getClient();
                 ClickHouseResponse response = newRequest(client, server)
-                        .query(ClickHouseParameterizedQuery.of(client.getConfig(), sql))
+                        .query(ClickHouseParameterizedQuery.of(client.getConfig(), sql), queryId)
                         .params(ClickHouseStringValue.of(uuid))
                         .executeAndWait()) {
             Assert.assertEquals(response.firstRecord().getValue(0).asString(), uuid);
         }
         ClickHouseClient.send(server, "SYSTEM FLUSH LOGS").get();
-        try (ClickHouseClient client = getClient();
-                ClickHouseResponse response = newRequest(client, server)
-                        .query("select http_user_agent from system.query_log where query='select ''" + uuid + "'''")
-                        .executeAndWait()) {
-            String result = response.firstRecord().getValue(0).asString();
-            Assert.assertTrue(result.startsWith(client.getConfig().getProductName()));
-            Assert.assertTrue(result.indexOf("Http") > 0);
+
+        String selectQuery = "select http_user_agent from system.query_log where query='select ''" + uuid + "'''";
+        if (isCloud()) {
+            selectQuery = "SELECT http_user_agent FROM clusterAllReplicas(default, 'system', query_log) WHERE query_id='" + queryId + "'";
         }
 
+        int loopCount = 0;
```

Comment:
The same as above: avoid mixing the test suite and service code. We can create a helper that wraps test execution and retries until a successful assertion or a boundary condition.
Consider reusing this helper in all the tests in the file